packages/talker_grpc_logger/lib/src/talker_grpc_logger_base.dart

```diff
@@ -0,0 +1,190 @@
+/// Implements a gRPC interceptor that logs requests and responses to Talker.
+/// https://pub.dev/documentation/grpc/latest/grpc/ClientInterceptor-class.html
+import 'dart:convert';
+
+import 'package:talker/talker.dart';
+import 'package:grpc/grpc.dart';
+
+const encoder = JsonEncoder.withIndent('  ');
+
+class TalkerGrpcLogger extends ClientInterceptor {
+  TalkerGrpcLogger({Talker? talker, this.obfuscateToken = true}) {
+    _talker = talker ?? Talker();
+  }
+
+  late Talker _talker;
+  final bool obfuscateToken;
+
+  @override
+  ResponseFuture<R> interceptUnary<Q, R>(ClientMethod<Q, R> method, Q request,
+      CallOptions options, ClientUnaryInvoker<Q, R> invoker) {
+    _talker.logTyped(GrpcRequestLog(method.path,
+        method: method,
+        request: request,
+        options: options,
+        obfuscateToken: obfuscateToken));
+
+    DateTime startTime = DateTime.now();
+    final response = invoker(method, request, options);
+
+    response.then((r) {
+      Duration elapsedTime = DateTime.now().difference(startTime);
+      _talker.logTyped(GrpcResponseLog(method.path,
+          method: method, response: r, durationMs: elapsedTime.inMilliseconds));
+    }).catchError((e) {
+      Duration elapsedTime = DateTime.now().difference(startTime);
+      _talker.logTyped(GrpcErrorLog(method.path,
+          method: method,
+          request: request,
+          options: options,
+          grpcError: e,
+          durationMs: elapsedTime.inMilliseconds,
+          obfuscateToken: obfuscateToken));
+    });
+    return response;
+  }
+
+  @override
+  ResponseStream<R> interceptStreaming<Q, R>(
+      ClientMethod<Q, R> method,
+      Stream<Q> requests,
+      CallOptions options,
+      ClientStreamingInvoker<Q, R> invoker) {
+    print('interceptStreaming');
+
+    return invoker(method, requests, options);
+  }
+}
+
+class GrpcRequestLog<Q, R> extends TalkerLog {
+  GrpcRequestLog(
```

Comment:
Can we move logs classes in different file like in talker_bloc_logger package ?

Reply:
done, moved to ` grpc_logs.dart`